pacman-nodejs-app/docs/pacman-nodejs-app-aws-gke-migration.md

```diff
@@ -0,0 +1,438 @@
+
+```bash
+kubectl get all -o wide
+```
+
+### Save Pac-Man and MongoDB Load Balancer IP
+
+```bash
+PACMAN_DST_PUBLIC_IP=$(kubectl get svc pacman --output jsonpath="{.status.loadBalancer.ingress[0].ip}")
+MONGO_DST_PUBLIC_IP=$(kubectl get svc mongo --output jsonpath="{.status.loadBalancer.ingress[0].ip}")
+```
+
+### Add New Mongo Instance to Replica Set
+
+Connect to the mongo pod in cluster A and invoke the mongo client to connect to the mongodb PRIMARY in order to add the new mongo instance as a replica set SECONDARY.
+
+```bash
+kubectl --context=${SUBDOMAIN} \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.add(\"${MONGO_DST_PUBLIC_IP}:27017\")"
+```
+
+### Check Status of New Mongo Instance In Replica Set
+
+```bash
+kubectl --context=${SUBDOMAIN} \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.status()"
+```
+
+### Make New Mongo Instance Primary
+
+```bash
+kubectl --context=${SUBDOMAIN} \
+    exec -it ${MONGO_SRC_POD} -- \
+    mongo --eval "rs.stepDown(120)"
+```
+
+### Update DNS
+
+Update DNS to point to new cluster:
+
+```bash
+gcloud dns record-sets transaction start -z=${ZONE_NAME}
+gcloud dns record-sets transaction remove -z=${ZONE_NAME} --name="pacman.${DNS_NAME}" --type=CNAME --ttl=1 "${PACMAN_SRC_PUBLIC_HOSTNAME}"
+gcloud dns record-sets transaction add -z=${ZONE_NAME} --name="pacman.${DNS_NAME}" --type=A --ttl=1 "${PACMAN_DST_PUBLIC_IP}"
+gcloud dns record-sets transaction execute -z=${ZONE_NAME}
+```
+
+### Remove Old Mongo Instance From Replica Set
+
+```bash
+MONGO_DST_POD=$(kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 get pod \
+    --selector="name=mongo" \
+    --output=jsonpath='{.items..metadata.name}')
+kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 \
+    exec -it ${MONGO_DST_POD} -- \
+    mongo --eval "rs.remove(\"${MONGO_SRC_PUBLIC_HOSTNAME}:27017\")"
+```
+
+### Check Status to Verify Removal
+
+```bash
+kubectl --context=gke_${GCP_PROJECT}_us-west1-b_gce-us-west1 \
+    exec -it ${MONGO_DST_POD} -- \
+    mongo --eval "rs.status()"
+```
+
+### Remove Old Pac-man Cluster Resources
+
+```bash
+kubectl --context ${SUBDOMAIN} \
+    delete ns pacman
+```
+
+## Migrate Pac-Man Application to Cluster B (Automated)
+
+### Prerequisites
+
+To perform this migration, the tool makes the following assumptions:
+
+1. AWS is used for cluster A and GKE is used for cluster B.
+2. Pac-Man is already deployed and working in cluster A. See above for steps.
+3. Both clusters are making use of Google Cloud DNS to manage DNS entries. [See here for more details](https://cloud.google.com/dns/migrating).
+4. Google Cloud DNS managed zone is already created in your Google Cloud Platform project.
+   [See here for instructions](kubernetes-cluster-gke-federation.md#cluster-dns-managed-zone).
+5. `gcloud` cloud SDK is installed and working on client to create the gke cluster and manage Google Cloud DNS. [See here for instructions](https://cloud.google.com/sdk/).
```

Comment:
s/gke/GKE/
What I originally meant here is that the tool doesn't care if `gcloud` is configured to create the GKE cluster as it only uses `gcloud` to manage Google Cloud DNS. So it doesn't make that assumption.

Reply:
Same comment